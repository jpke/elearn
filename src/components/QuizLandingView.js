import React from 'react';
import {Link} from 'react-router';
import QuizContainer from '../containers/QuizContainer';
import QuizStartContainer from '../containers/QuizStartContainer';
import QuizListContainer from '../containers/QuizListContainer';

//view for QuizLandingContainer
//displays message redirecting to AuthContainer when user is not logged in
//displays QuizViewContainer when user has started quiz
//displays QuizStartContainer when user has selected quiz but not yet started it
//displays QuizListContainer otherwise
const QuizLandingView = (props) => {
  return (
    <div className="quizContainer">
      <h2>{props.courseName ? props.courseName + " Quizzes" : "Quizzes"}</h2>
      {props.token ?
        //if user is authenticated check for if quiz is in progress
      (
        props.quizInProgress ?
        //if so, display quiz
          <QuizContainer />
         :
         //otherwise check if a quiz has been selected from the list
         props.viewQuizSelected ?
          //if so, display start page for that quiz, which also displays user stats for that quiz
          <QuizStartContainer />
          :
          //otherwise display list of quizzes available
          //if user is course admin, this list offers option to go to edit quiz view
          <QuizListContainer />
        ) : (
        <h3>You must be <Link to='/' className="redirect">logged in</Link> to access this content</h3>
        )
      }
    </div>
  );
}

export default QuizLandingView;
