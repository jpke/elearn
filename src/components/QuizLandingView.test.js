/* eslint-disable no-undef */
/* eslint-disable import/no-named-as-default */

import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from '../store/configureStore';
const store = configureStore();

import QuizLandingView from './QuizLandingView';

it('renders correctly for logged out user', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <QuizLandingView
        viewQuizSelected={false}
        courseName={"sample course name"}
        quizInProgress={false}
        token={""}
      />
  </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly for logged in user with unselected quiz', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <QuizLandingView
        viewQuizSelected={false}
        courseName={"sample course name"}
        quizInProgress={false}
        token={"0000"}
      />
  </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly for logged in user with selected quiz not started', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <QuizLandingView
        viewQuizSelected={true}
        courseName={"sample course name"}
        quizInProgress={false}
        token={"0000"}
      />
  </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly for logged in user with selected quiz started', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <QuizLandingView
        viewQuizSelected={true}
        courseName={"sample course name"}
        quizInProgress={true}
        token={"0000"}
      />
  </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
