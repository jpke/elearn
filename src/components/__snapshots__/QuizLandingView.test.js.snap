exports[`test renders correctly for logged in user with selected quiz not started 1`] = `
<div
  className="quizContainer">
  <h2>
    sample course name Quizzes
  </h2>
  <div
    className="quiz-start">
    <h3>
      You have not yet attempted this quiz
    </h3>
    <div
      className="quiz-start-button-container">
      <button
        className="quiz-start-button"
        onClick={[Function]}>
         Start Quiz 
      </button>
      <button
        className="quiz-start-button"
        onClick={[Function]}>
        Back to Quiz List
      </button>
    </div>
  </div>
</div>
`;

exports[`test renders correctly for logged in user with selected quiz started 1`] = `
<div
  className="quizContainer">
  <h2>
    sample course name Quizzes
  </h2>
  <div
    className="quizInProgress">
    <h3
      className="title">
      new quiz
    </h3>
    <h3>
      Question 
      1
    </h3>
    <p>
      no current question
    </p>
    <h3
      className="answers">
      Answers
    </h3>
    <ul
      className="item-list">
      <div>
        <li
          className="none"
          id={undefined}
          onClick={[Function]}>
          no current answer
        </li>
        <li
          className="none"
          id={undefined}
          onClick={[Function]}>
          no current answer true
        </li>
      </div>
    </ul>
    <div
      className="quizNav-buttons">
      <button
        onClick={[Function]}>
        Next Question
      </button>
    </div>
  </div>
</div>
`;

exports[`test renders correctly for logged in user with unselected quiz 1`] = `
<div
  className="quizContainer">
  <h2>
    sample course name Quizzes
  </h2>
  <div>
    <h3>
      Select quiz
    </h3>
    <div
      className="quiz-list-container">
      <ul
        className="quiz-list">
        <li>
          No quizzes listed 
          <br />
           Have you selected a course?
        </li>
      </ul>
      <ul
        className="passed" />
    </div>
  </div>
</div>
`;

exports[`test renders correctly for logged out user 1`] = `
<div
  className="quizContainer">
  <h2>
    sample course name Quizzes
  </h2>
  <h3>
    You must be 
    <a
      className="redirect"
      onClick={[Function]}
      style={Object {}}>
      logged in
    </a>
     to access this content
  </h3>
</div>
`;
